db/keyval/filedb/decoder/decoder_yaml.go

```diff
@@ -0,0 +1,89 @@
+//  Copyright (c) 2018 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package decoder
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/ghodss/yaml"
+	"strings"
+)
+
+// Extension supported by this decoder
+var yamlDcExtensions = []string{".yaml"}
```

Comment:
This could be defined as exported field in `YAMLDecoder` to allow customization.